File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/config/KafkaTopicProperties.java
package com.paulopacifico.orderservice.messaging.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "app.kafka.topics")
public record KafkaTopicProperties(
        String orderPlaced,
        String inventoryReserved,
        String inventoryFailed,
        String orderFailed,
        String orderConfirmed,
        String paymentSucceeded,
        String paymentFailed,
        String orderPaid,
        String shipmentShipped,
        String shipmentFailed
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/config/KafkaTopicProperties.kt
package com.paulopacifico.inventoryservice.messaging.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("app.kafka.topics")
data class KafkaTopicProperties(
    var orderPlaced: String = "order-placed-topic",
    var inventoryReserved: String = "inventory-reserved-topic",
    var inventoryFailed: String = "inventory-failed-topic",
    var orderFailed: String = "order-failed-topic",
    var paymentFailed: String = "payment-failed-topic",
)
